maven-surefire-common/src/main/java/org/apache/maven/surefire/providerapi/ServiceLoader.java

```diff
@@ -42,7 +45,8 @@
  *
  * @since 2.20
  */
-public final class ServiceLoader
+@Component( role = ServiceLoader.class )
+public class ServiceLoader
```

Comment:
@slawekjaranowski Can this Component be also `public final` class as the other classes in this package? So that we indicate that the component class cannot be extended by the user in another project.

Reply:
I use mock `ServiceLoader` of in `ProviderDetectorTest` but final clases can't be mocked.
I can introduce interfaces for components and mock on interface will be posible.

Reply:
ah ok, so let it be, the tests with mock are more important